### PR TITLE
Add support for new Low Life and Full Life Masteries

### PR DESCRIPTION
**Pull request by Nostrademous**
Made ConfigOption tooltips dynamically fillable (by turning into tooltipFuncs).
Auto set Full Life and Low Life appropriately based on masteries.

Build Tested with:
https://pobb.in/7bek_xBstkpA

### CODE REVIEW

File: src/Modules/CalcPerform.lua [hunk 1]
```diff
@@ -674,8 +674,21 @@ local function doActorAttribsPoolsConditions(env, actor)
...
+		data.misc.LowLifePercentage = lowLifePerc
+	else
+		data.misc.LowLifePercentage = 50.0
+	end
+	local fullLifePerc = modDB:Sum("BASE", nil, "FullLifePercentage")
+	if fullLifePerc > 0 then
+		data.misc.FullLifePercentage = fullLifePerc
+	else
+		data.misc.FullLifePercentage = 100.0
+	end
```
Comment (Wires77) [inline src/Modules/CalcPerform.lua:689]: Don't change things in the data file during calculation.  That consists of static data that is loaded on start-up.

File: src/Modules/CalcPerform.lua [hunk 2]
```diff
@@ -742,9 +755,12 @@ local function doActorLifeManaReservation(actor)
 			output[pool.."ReservedPercent"] = m_min(reserved / max * 100, 100)
 			output[pool.."Unreserved"] = max - reserved
 			output[pool.."UnreservedPercent"] = (max - reserved) / max * 100
-			if (max - reserved) / max <= data.misc.LowPoolThreshold then
+			if (max - reserved) / max <= (pool == "Life" and data.misc.LowLifePercentage / 100.0 or data.misc.LowPoolThreshold) then
```
Comment (Wires77) [inline src/Modules/CalcPerform.lua:758]: No need to break the generic parts of this loop.  Do something like this (maybe pull the Sum out into a local variable):
```suggestion
			if (max - reserved) / max <= modDB:Sum("BASE", nil, "Low" .. pool .. "Percentage") / 100.0 or data.misc.LowPoolThreshold) then
```

File: src/Modules/ModParser.lua [hunk 3]
```diff
@@ -3172,7 +3172,8 @@ local specialModList = {
 		mod("ShrineBuff", "LIST", { mod = mod("Life", "INC", 20) }),
 		mod("ShrineBuff", "LIST", { mod = mod("AreaOfEffect", "INC", 20) })
 	},
-	-- 
+	["you count as on low life while at (%d+)%% of maximum life or below"] = function(num) return { mod("LowLifePercentage", "BASE", num) } end,
+	["you count as on full life while at (%d+)%% of maximum life or above"] = function(num) return { mod("FullLifePercentage", "BASE", num) } end,
```
Comment (Wires77) [inline src/Modules/ModParser.lua:3176]: Want to future-proof these and add 'life' as a parameter too?  Then we don't need a PR for if they ever add mana related mods like this.

**Review by Wires77 — CHANGES_REQUESTED** [review]
You'll have to fix a merge conflict, since this block got moved, but this could be made a lot more generic to both future proof and help code readability.

File: src/Modules/CalcPerform.lua (hunk 1 above)
Comment (Nostrademous) [reply src/Modules/CalcPerform.lua:689]: @Wires77 how do I pass non-data file values into `ConfigOptions.lua`?

Comment (Wires77) [reply src/Modules/CalcPerform.lua:689]: Grr, you're right, you can't.  This is fine then, even if I hate it from a clean code perspective 🥲 It's just not worth writing a bunch of code to bring build data into the configOptions

**Review by Wires77 — APPROVED** [review]

**Pull request merged**